includes/class-wc-payments-checkout.php

```diff
@@ -207,6 +207,7 @@ public function get_payment_fields_js_config() {
 			'isWooPayGlobalThemeSupportEnabled' => $this->gateway->is_woopay_global_theme_support_enabled(),
 			'woopayHost'                        => WooPay_Utilities::get_woopay_url(),
 			'platformTrackerNonce'              => wp_create_nonce( 'platform_tracks_nonce' ),
+			'enableShopperTracking'             => apply_filters( 'wcpay_enable_shopper_tracking', true ) && 'no' !== get_option( 'woocommerce_allow_tracking' ),
```

Comment:
(minor) since this is a new filter, what do you think of doing something like
```suggestion
			'enableShopperTracking'             => apply_filters( 'wcpay_enable_shopper_tracking', 'no' !== get_option( 'woocommerce_allow_tracking' ) ),
```

This way, the filter will still honor WC Core's option value.

Reply:
addressed here a794891fff06d2d541ac4c03b11cf6896020bfbb